src/core/ext/transport/chttp2/server/insecure/server_chttp2_posix.c

```diff
@@ -57,6 +59,9 @@ void grpc_server_add_insecure_channel_from_fd(grpc_server *server,
   char *name;
   gpr_asprintf(&name, "fd:%d", fd);
 
+  int flags = fcntl(fd, F_GETFL, 0);
```

Comment:
It might be better to do:

```
GPR_ASSERT(GRPC_LOG_IF_ERROR("set_socket_non_blocking", grpc_set_socket_non_blocking(fd, true)));
```

Failure will generate a better error message at least, and later on we'll be able to support recovery a little easier.